src/backend/distributed/planner/merge_planner.c

```diff
@@ -849,6 +869,15 @@ ConvertRelationRTEIntoSubquery(Query *mergeQuery, RangeTblEntry *sourceRte,
 
 	/* replace the function with the constructed subquery */
 	sourceRte->rtekind = RTE_SUBQUERY;
+#if PG_VERSION_NUM >= PG_VERSION_16
+	sourceRte->perminfoindex = 0;
+
+	/*
+	 * Note: we don't need to remove replaced sourceRte from mergeQuery->rteperminfos to avoid
```

Comment:
How is this comment accurate? We are in the planner, and ExecCheckPermissions is on the executor, so we are definitely not gone through ExecCheckPermissions.

Do we really need these comments? They seem like development artifacts to me.